torch/distributed/pipelining/schedules.py

```diff
@@ -11,7 +11,7 @@
 from collections.abc import Callable
 from enum import Enum
 from functools import lru_cache
-from typing import Any, NamedTuple, Optional, Union
+from typing import Any, Callable, NamedTuple, Optional, Protocol, Union
```

Comment:
`typing.Callable` is deprecated since 3.10. Consider use `collections.abc.Callable`.